bridgetown-core/lib/bridgetown-core/front_matter/loaders/base.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Bridgetown
+  module FrontMatter
+    module Loaders
+      # An abstract base class for processing front matter
+      class Base
+        # @param origin_or_layout [Bridgetown::Model::RepoOrigin, Bridgetown::Layout]
+        def initialize(origin_or_layout)
+          @origin_or_layout = origin_or_layout
```

Comment:
**comment:** I don't know if there's a higher-level concept that these things both have in common. Is there?